feat: implement TransactionRepository for Hedera Mirror Node integration

belloibrahv opened the pull request:
## Overview
This PR implements the `TransactionRepository` interface and its implementation to interact with Hedera's Mirror Node REST API for transaction-related operations. The implementation follows the existing pattern established by the `NftRepository`.

## Changes
- Added `TransactionRepository` interface with methods to query transactions
- Implemented `TransactionRepositoryImpl` class
- Extended `MirrorNodeClient` with transaction-related query methods

### Key Components

#### TransactionRepository Interface
- Defines two primary methods:
  - `findByAccount`: Retrieves all transactions for a specific account
  - `findById`: Retrieves a specific transaction by its ID
- Includes a default method for string-based account IDs for convenience

#### TransactionRepositoryImpl
- Implements the repository interface using `MirrorNodeClient`
- Follows the same pattern as `NftRepositoryImpl`
- Includes proper null checks for all parameters
- Utilizes dependency injection for `MirrorNodeClient`

#### MirrorNodeClient Extensions
- Added new methods to support transaction queries:
  - `queryTransactionsByAccount`
  - `queryTransaction`
- These methods align with the Mirror Node REST API endpoints:
  - `/api/v1/transactions`
  - `/api/v1/transactions/{transactionId}`

## Design Decisions
1. Followed the established repository pattern from `NftRepository`
2. Kept the interface focused on core transaction operations
3. Used `Optional` for single transaction queries to handle cases where transactions might not exist
4. Implemented default methods for string-based IDs to improve usability

## Testing Note
Tests are currently not included as I couldn't find existing test patterns in the hedera-base codebase. Would appreciate reviewer guidance on testing approach and expectations.

## Related Issues
Closes #39

## Questions for Reviewers
-  What is the preferred testing approach for repository implementations?

hendrikebbers commented:
@belloibrahv that looks very good. But if you look at the broken build (https://github.com/OpenElements/hedera-enterprise/actions/runs/11609044311/job/32470469936?pr=61) you can see that it ends/fails with the following error:

```
/home/runner/work/hedera-enterprise/hedera-enterprise/hedera-spring/src/main/java/com/openelements/hedera/spring/implementation/MirrorNodeClientImpl.java:[31,8] com.openelements.hedera.spring.implementation.MirrorNodeClientImpl is not abstract and does not override abstract method queryTransactionsByAccount(com.hedera.hashgraph.sdk.AccountId) in com.openelements.hedera.base.mirrornode.MirrorNodeClient
```

While you added the new method to `com.openelements.hedera.base.mirrornode.MirrorNodeClient` you missed to add an implementation to `MirrorNodeClientImpl.java`.

belloibrahv commented:
Hi  @hendrikebbers,

I've implemented the missing `queryTransactionsByAccount` method in the `MirrorNodeClientImpl` class. This should address the issue with the broken build reported in the previous PR.

The key changes are:

1. Implemented the `queryTransactionsByAccount` method, which constructs the API endpoint path and uses the `RestBasedPage` class to make the API call and extract the list of `TransactionInfo` objects.
2. Added a private `extractTransactionInfoFromJsonNode` method to handle the parsing of the JSON response and creation of `TransactionInfo` objects.

I've tested the changes locally and the unit tests are passing. However, I'd appreciate if you could review the implementation and provide any feedback. Also the actual for `TransactionRepositoryImpl `  Tests are currently not included as I couldn't find existing test patterns in the hedera-base codebase. Would appreciate reviewer guidance on testing approach and expectations.


Please let me know if you have any questions or concerns.

Thank you!

hendrikebbers commented:
@belloibrahv that looks good :) I will execute the tests and if all is working I will merge it :)
